source/server/http/admin.cc

```diff
@@ -518,43 +520,65 @@ AdminImpl::statsAsJson(const std::map<std::string, uint64_t>& all_stats,
+  Value histograms_container_obj;
+  histograms_container_obj.SetObject();
+
+  Value histograms_obj;
+  histograms_obj.SetObject();
+
+  // It is not possible for the supported quantiles to differ across histograms, so it is ok to
+  // send them once.
+  Stats::HistogramStatisticsImpl empty_statistics;
+  rapidjson::Value supported_quantile_array(rapidjson::kArrayType);
+  for (double quantile : empty_statistics.supportedQuantiles()) {
+    Value quantile_type;
+    quantile_type.SetDouble(quantile * 100);
+    supported_quantile_array.PushBack(quantile_type, allocator);
+  }
+  histograms_obj.AddMember("supported_quantiles", supported_quantile_array, allocator);
+  rapidjson::Value histogram_array(rapidjson::kArrayType);
+
   for (const Stats::ParentHistogramSharedPtr& histogram : all_histograms) {
-    Value histogram_obj;
-    histogram_obj.SetObject();
-    Value histogram_name;
-    histogram_name.SetString(histogram->name().c_str(), allocator);
-    histogram_obj.AddMember("name", histogram_name, allocator);
-
-    rapidjson::Value quantile_array(rapidjson::kArrayType);
-
-    // TODO(ramaraochavali): consider optimizing the model here. Quantiles can be added once,
-    // followed by two arrays interval and cumulative.
-    for (size_t i = 0; i < histogram->intervalStatistics().supportedQuantiles().size(); ++i) {
-      Value quantile_obj;
-      quantile_obj.SetObject();
-      Value quantile_type;
-      quantile_type.SetDouble(histogram->intervalStatistics().supportedQuantiles()[i] * 100);
-      quantile_obj.AddMember("quantile", quantile_type, allocator);
-      Value interval_value;
-      if (!std::isnan(histogram->intervalStatistics().computedQuantiles()[i])) {
-        interval_value.SetDouble(histogram->intervalStatistics().computedQuantiles()[i]);
-      }
-      quantile_obj.AddMember("interval_value", interval_value, allocator);
-      Value cumulative_value;
-      if (!std::isnan(histogram->cumulativeStatistics().computedQuantiles()[i])) {
-        cumulative_value.SetDouble(histogram->cumulativeStatistics().computedQuantiles()[i]);
+    if (histogram->used()) {
+      Value histogram_obj;
+      histogram_obj.SetObject();
+      Value histogram_name;
+      histogram_name.SetString(histogram->name().c_str(), allocator);
+      histogram_obj.AddMember("name", histogram_name, allocator);
+
+      rapidjson::Value computed_quantile_array(rapidjson::kArrayType);
+
+      for (size_t i = 0; i < histogram->intervalStatistics().supportedQuantiles().size(); ++i) {
+        Value quantile_obj;
+        quantile_obj.SetObject();
+        Value interval_value;
+        if (!std::isnan(histogram->intervalStatistics().computedQuantiles()[i])) {
+          interval_value.SetDouble(histogram->intervalStatistics().computedQuantiles()[i]);
+        }
+        quantile_obj.AddMember("interval", interval_value, allocator);
+        Value cumulative_value;
+        if (!std::isnan(histogram->cumulativeStatistics().computedQuantiles()[i])) {
```

Comment:
Probably worth commenting here that as we are positionally associating  this entry with a quantile, so we need to put in the {null, null} entry to keep other data aligned.